cobigen-cli/src/main/java/com/devonfw/cobigen/cli/utils/ParsingUtils.java

```diff
@@ -57,19 +57,49 @@
      */
     public static JavaContext getJavaContext(File inputFile, File inputProject) {
 
-        JavaContext context = JavaSourceProviderUsingMaven.createFromLocalMavenProject(inputProject);
-
+        JavaContext context = JavaSourceProviderUsingMaven.createFromLocalMavenProject(inputProject,true);
+        String fqn = ParsingUtils.getFQN(inputFile);
         String qualifiedName = ParsingUtils.getQualifiedName(inputFile, context);
 
         try {
-            context.getClassLoader().loadClass(qualifiedName);
+           // context.getClassLoader().loadClass(qualifiedName);
+        	 context.getClassLoader().loadClass(fqn);
         } catch (NoClassDefFoundError | ClassNotFoundException e) {
             logger.error("Compiled class " + e.getMessage()
                 + " has not been found. Most probably you need to build project " + inputProject.toString() + " .");
             System.exit(1);
         }
         return context;
     }
+    /**
+     * @param inputFile
+     * @return
+     */
+    private static String getFQN(File inputFile) {
+        String simpleName = inputFile.getName().replaceAll("\\.(?i)java", "");
+        String packageName = getPackageName(inputFile.getParentFile(), "");
+
+         return packageName + "." + simpleName;
+    }
+
+     private static String getPackageName(File folder, String packageName) {
```

Comment:
Can you add Javadoc? This method is traversing parent folders until it reaches `java` folder in order to get the FQN